src/go/rpk/pkg/cli/cmd/cluster/maintenance/enable.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package maintenance
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newEnableCommand(fs afero.Fs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "enable <node-id>",
+		Short: "Enable maintenance mode for a node.",
+		Long: `Enable maintenance mode for a node.
+
+This command enables maintenance mode for the node with the specified ID. If a
+node exists that is already in maintenance mode then an error will be returned.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			node_id, err := strconv.Atoi(args[0])
+			if err != nil {
+				out.MaybeDie(err, "could not parse node id: %s: %v", args[0], err)
+			}
+
+			if node_id < 0 {
+				out.Die("invalid node id: %d", node_id)
+			}
+
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			client, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			// check if broker exists
+			broker_found := false
+			brokers, err := client.Brokers()
+			for _, broker := range brokers {
+				if broker.NodeID == node_id {
+					broker_found = true
+					break
+				}
+			}
+
+			if !broker_found {
+				out.Die("Unknown node %d", node_id)
+			}
+
+			err = client.EnableMaintenanceMode(node_id)
+			out.MaybeDie(err, "error enabling maintenance mode for node %d: %v", node_id, err)
```

Comment:
Did you check that this returns a reasonably clear error in cases like another node already being in maintenance mode?  I'm not sure if the %v of the error is just going to be the request body, or some sort of curly-brace-filled thing.

Reply:
It does in fact return a gross thing. I changed this to now use your approach in config.go etc... where it looks like the strategy is to provide nice messages where possible and then at the very end fall back printing out whatever the raw body as a last resort. So now for a few common cases it prints out a nice error message. Does that sound about right?